packages/pwa/app/utils/url.js

```diff
@@ -102,16 +103,23 @@ export const productUrlBuilder = (product) => encodeURI(`/product/${product.id}`
 export const searchUrlBuilder = (searchTerm) => `/search?q=${searchTerm}`
 
 /**
- * Returns a URL using the new locale.
+ * Returns a URL pathname using the new locale.
  *
- * @param previousLocale
- * @param newLocale
+ * @param {string} newLocale
+ * @param {Object} [options]
+ * @param {Object} options.location - original location (similar value to `window.location`)
  * @returns {string}
  */
-export const buildUrlLocale = (previousLocale, newLocale) => {
+export const buildUrlLocale = (newLocale, options = {}) => {
+    const location = options.location
+        ? options.location
+        : typeof window !== 'undefined'
```

Comment:
Should we prioritize `useLocation` than using window.location as we are using react-router-dom? 🤔

Reply:
I had to prioritize `window.location` because the function is used in these 2 scenarios:
- inside an event handler (for the locale selector dropdown) ← in this case, `window.location` is used
- inside React component ← in this case, the `useLocation` hook is called

Let me add some comments to make it clearer. If perhaps there's a clearer way of writing/organizing the code, let me know too please.